WordPress/Classes/Services/MediaService.m

```diff
@@ -242,14 +255,6 @@ - (void)uploadMedia:(Media *)media
         return;
     }
 
-    [self.managedObjectContext performBlock:^{
```

Comment:
The original issue wasn't that though.  The issue I was having is that after calling `uploadMedia()`, a subsequent CD query could still return that media object as still being `.local` instead of `.pushing`, thus not recognizing there was an upload in progress for the media object.

Reply:
@diegoreymendez Do you have some reproduction steps for that so I can test if this PR fixes the problem?

Reply:
Basically this is what's causing issues with test case 5 here: https://github.com/wordpress-mobile/WordPress-iOS/pull/12416

The closest thing I can provide to being able to test the issue is to follow the testing steps detailed in this PR against `develop` and once you hit step 6 notice the `remoteStatus` for the post is still `.local`.

You can try: `po MediaCoordinator.shared.isUploadingMedia(for: post)` which will return `false` (this method checks the `remoteStatus` of posts).